editor/plugins/script_editor_plugin.cpp

```diff
@@ -3463,15 +3471,16 @@ void ScriptEditorPlugin::edit(Object *p_object) {
 	if (Object::cast_to<Script>(p_object)) {
 
 		Script *p_script = Object::cast_to<Script>(p_object);
-		String scene_path = p_script->get_path().get_slice("::", 0);
-
-		if (_is_built_in_script(p_script) && !EditorNode::get_singleton()->is_scene_open(scene_path)) {
-			EditorNode::get_singleton()->load_scene(scene_path);
+		String res_path = p_script->get_path().get_slice("::", 0);
 
-			script_editor->call_deferred("edit", p_script);
-		} else {
-			script_editor->edit(p_script);
+		if (_is_built_in_script(p_script) && !EditorNode::get_singleton()->is_scene_open(res_path)) {
+			if (ResourceLoader::get_resource_type(res_path) == "PackedScene") {
+				EditorNode::get_singleton()->load_scene(res_path);
+			} else {
+				EditorNode::get_singleton()->load_resource(res_path);
+			}
```

Comment:
The if's here should align with the other ones, checking for the open scene after it's confirmed to be a `PackedScene`.

Reply:
Done!